test/e2e/data/infrastructure-aws/kustomize_sources/topology/clusterclass-quick-start.yaml

```diff
@@ -158,6 +171,49 @@ spec:
             path: "/spec/template/spec/controlPlaneLoadBalancer/healthCheckProtocol"
             valueFrom:
               variable: healthCheckProtocol
+  - name: preKubeadmCommands
+    enabledIf: '{{ eq .selfHosted "yes" }}'
+    definitions:
+      - selector:
+          apiVersion: controlplane.cluster.x-k8s.io/v1beta1
+          kind: KubeadmControlPlaneTemplate
+          matchResources:
+            controlPlane: true
+        jsonPatches:
+          - op: add
+            path: "/spec/template/spec/kubeadmConfigSpec/preKubeadmCommands"
+            valueFrom:
+              template: |
+                - mkdir -p /opt/cluster-api
+                - ctr -n k8s.io images pull "${CAPI_IMAGES_REGISTRY}:${E2E_IMAGE_TAG}"
+                - ctr -n k8s.io images tag "${CAPI_IMAGES_REGISTRY}:${E2E_IMAGE_TAG}" gcr.io/k8s-staging-cluster-api/capa-manager:e2e
```

Comment:
Do we need this in quick start test?

Reply:
A good question. I also debated about the approach. 

The current approach works as the preKubeadmCommands is patched only when `selfHosted` variable is set to yes. Another approach would be to create a separate clusterclass (not quickstart clusterclass) for the self-hosted case if it makes things cleaner.

I don't have a strong opinion and am ok with either approach. Let me know what you think.

Reply:
Better to have one CC IMO. If this file is shared between tests, then how about changing its name to something more generic and drop test specific postfix?

Reply:
Sounds great. how about clusterclass-default or clusterclass-ci? 
Since all test passed, let's merge this PR. I will create a follow up PR updating the name and refactor the references.